scripts/ci/create-docker-image.sh

```diff
@@ -21,68 +21,80 @@ if [ "${DOCKER_IMAGE%/*/*}" != "${DOCKER_IMAGE}" ]; then
             ;;
           s390x)
-            make docker-cross-build WITH_EBPF=true TARGET_ARCH=s390x TARGET_GOARCH=s390x DOCKER_IMAGE=${DOCKER_IMAGE} BASE=${BASE:-${arch}/ubuntu} DOCKER_TAG=$arch-${DOCKER_TAG}
+            make docker-cross-build ${flags} WITH_EBPF=true TARGET_ARCH=$arch TARGET_GOARCH=$arch BASE=${BASE:-${arch}/ubuntu}
             ;;
           *)
-            make docker-cross-build WITH_EBPF=true TARGET_ARCH=$arch TARGET_GOARCH=$arch DOCKER_IMAGE=${DOCKER_IMAGE} BASE=${BASE:-${arch}/ubuntu} DOCKER_TAG=$arch-${DOCKER_TAG}
+            make docker-cross-build ${flags} WITH_EBPF=true TARGET_ARCH=$arch TARGET_GOARCH=$arch BASE=${BASE:-${arch}/ubuntu}
             ;;
         esac
     done
-fi
+}
 
-if [ -n "$DRY_RUN" ]; then
-    echo "Running in dry run mode. Exiting."
-    exit 0
-fi
+docker_login() {
+    set +x
+    if [ -z "$DOCKER_PASSWORD" ]; then
+        echo "The environment variable DOCKER_PASSWORD needs to be defined"
+        exit 1
+    fi
 
-set +x
-if [ -z "$DOCKER_PASSWORD" ]; then
-    echo "The environment variable DOCKER_PASSWORD needs to be defined"
-    exit 1
-fi
+    echo "${DOCKER_PASSWORD}" | docker login  --username "${DOCKER_USERNAME}" --password-stdin ${DOCKER_SERVER}
+    set -x
+}
 
-echo "${DOCKER_PASSWORD}" | docker login  --username "${DOCKER_USERNAME}" --password-stdin ${DOCKER_SERVER}
-set -x
+docker_push() {
+    for arch in $ARCHES
+    do
+        docker push ${DOCKER_IMAGE}:${arch}-${DOCKER_TAG}
+    done
+}
 
-digests=""
-for arch in $ARCHES
-do
-    docker push ${DOCKER_IMAGE}:${arch}-${DOCKER_TAG}
+docker_manifest() {
+    digests=""
+    for arch in $ARCHES
+    do
+        digest=$( docker inspect --format='{{index .RepoDigests 0}}' ${DOCKER_IMAGE}:${arch}-${DOCKER_TAG} )
+        digests="${digests} $digest"
+    done
 
-    digest=$( docker inspect --format='{{index .RepoDigests 0}}' ${DOCKER_IMAGE}:${arch}-${DOCKER_TAG} )
-    digests="${digests} $digest"
-done
+    res=0
+    for i in {1..6}
+    do
+        docker manifest create --amend "${DOCKER_IMAGE}:${DOCKER_TAG}" ${digests} && break || res=$?
+        sleep 10
+    done
+    [ $res != 0 ] && exit $res
 
-res=0
-for i in {1..6}
-do
-    docker manifest create --amend "${DOCKER_IMAGE}:${DOCKER_TAG}" ${digests} && break || res=$?
-    sleep 10
-done
-[ $res != 0 ] && exit $res
+    for arch in $ARCHES
+    do
+        digest=$( docker inspect --format='{{index .RepoDigests 0}}' ${DOCKER_IMAGE}:${arch}-${DOCKER_TAG} )
+        docker manifest annotate --arch $arch "${DOCKER_IMAGE}:${DOCKER_TAG}" $digest
+    done
 
-for arch in $ARCHES
-do
-    digest=$( docker inspect --format='{{index .RepoDigests 0}}' ${DOCKER_IMAGE}:${arch}-${DOCKER_TAG} )
-    docker manifest annotate --arch $arch "${DOCKER_IMAGE}:${DOCKER_TAG}" $digest
-done
+    docker manifest inspect "${DOCKER_IMAGE}:${DOCKER_TAG}"
+    docker manifest push --purge "${DOCKER_IMAGE}:${DOCKER_TAG}"
+}
+
+if [ -n "$DRY_RUN" ]; then
```

Comment:
The goal of DRY_RUN is to build the image but not push it. So this should be move after the `docker_build`

Reply:
ok